pinot-common/src/main/java/org/apache/pinot/common/datablock/BaseDataBlock.java

```diff
@@ -680,29 +680,4 @@ public String toString() {
     }
     return stringBuilder.toString();
   }
```

Comment:
since BaseDataBlock doesn't use ANY of the DataTable API. another way to refactor is to completely decoupled DataTable and DataBlock into 2 separate construct. 

--> now that https://github.com/apache/pinot/pull/9561/ is in, we can actually do that.